src/pages/products/products.jsx

```diff
@@ -3,146 +3,89 @@ import axios from "axios";
 import { Footer } from "../../components/footer/footer";
 import { Navbar } from "../../components/navbar/navbar";
 import "./products.css";
+import { Filter } from "../../components/Filter/filter";
+import { useFilter } from "../../context/filtercontext";
+
+import {
+  ratingItems,
+  categoryItems,
+  priceItems,
+  sortItems,
+} from "../../utils/index";
 
 export const Products = () => {
   const [data, setData] = useState([]);
   const [load, setLoad] = useState(false);
 
   useEffect(() => {
     loader();
-  }, []);
+  });
 
   const loader = async () => {
```

Comment:
Here the function name loader does not make any sense i think your are fetching product data from api so give a more meaningful name.